processor/lsmintervalprocessor/internal/merger/merger.go

```diff
@@ -21,35 +21,44 @@ import (
 	"io"
 
 	"github.com/cockroachdb/pebble"
+	"github.com/elastic/opentelemetry-collector-components/processor/lsmintervalprocessor/config"
 )
 
 var _ pebble.ValueMerger = (*Merger)(nil)
 
 type Merger struct {
-	current Value
+	current          Value
```

Comment:
Now that Merger methods have non-pointer receivers, they'll be copying the internals. Won't copying Value be pretty expensive?

Reply:
> Won't copying Value be pretty expensive?

Hmm, not by much I think because most of the value is reference type. Even pmetric is kinda wrapper around the proto OTel data model references. Though, I didn't intend to change the merger to value receiver but only add the limit config explicitly 😅 I think this happened by mistake, I will fix this.